examples/ssh/macos_disk_partition.js

```diff
@@ -0,0 +1,205 @@
+  if (bytes >= 1e9) {
+    return {
+      value: (bytes / 1e9).toFixed(2),
+      unit: 'GB'
+    }
+  } else {
+    return {
+      value: (bytes / 1e6).toFixed(2),
+      unit: 'MB'
+    }
+  }
+}
+
+/**
+ * Extracts relevant variables from parsed disk partitons and populates the table
+ * @param {Array} partitions Array of objects representing parsed disk partitonsinformation
+ */
+function populateTable(partitions) {
+  var parentDiskValues = {}
+  var table
+  partitions.forEach(function (disk) {
+    var containerTotalSpace = disk['Container Total Space'] ? disk['Container Total Space'].match(/(\d+) Bytes/) : 0
+    var volumeTotalSpace = disk['Volume Total Space'] ? disk['Volume Total Space'].match(/(\d+) Bytes/) : 0;
+    var totalSpaceBytes = containerTotalSpace ? parseInt(containerTotalSpace[1]) : (volumeTotalSpace ? parseInt(volumeTotalSpace[1]) : 0)
+    var containerFreeSpace = disk['Container Free Space'] ? disk['Container Free Space'].match(/(\d+) Bytes/) : 0
+    var volumeFreeSpace = disk['Volume Free Space'] ? disk['Volume Free Space'].match(/(\d+) Bytes/) : 0
+    var totalFreeSpaceBytes = containerFreeSpace ? parseInt(containerFreeSpace[1]) : (volumeFreeSpace ? parseInt(volumeFreeSpace[1]) : 0)
+    var totalUsedSpace = totalSpaceBytes - totalFreeSpaceBytes
+    var percentageUsedSpace = totalSpaceBytes > 0 ? (totalUsedSpace / totalSpaceBytes) * 100 : 0
+    var parentDisk = disk['Part of Whole']
+    if (!parentDiskValues[parentDisk]) {
+      var freeSpace = convertToAppropriateUnits(totalFreeSpaceBytes)
+      var size = convertToAppropriateUnits(totalSpaceBytes)
+      parentDiskValues[parentDisk] = {
+        freeSpace: freeSpace.value,
+        size: size.value,
+        freeSpaceUnit: freeSpace.unit,
+        sizeUnit: size.unit
+      }
+      if (!table) {
+        table = D.createTable(
+          "Disk partitions", 
+          [
+            { label: "Usage", unit: '%', valueType: D.valueType.NUMBER },
+            { label: "Free Space", unit: parentDiskValues[parentDisk].freeSpaceUnit, valueType: D.valueType.NUMBER },
+            { label: "Size", unit: parentDiskValues[parentDisk].sizeUnit, valueType: D.valueType.NUMBER }
+          ]
+        )
+      }
+      table.insertRecord(sanitize(parentDisk), 
+        [
+          percentageUsedSpace,
+          parentDiskValues[parentDisk].freeSpace,
+          parentDiskValues[parentDisk].size
+        ]
+      )
+    }
+  })
+  D.success(table)
+}
+
+/**
+ * @remote_procedure
+ * @label Validate SSH connectivity with the device
+ * @documentation This procedure is used to validate the driver and credentials provided during association.
+ */
+function validate () {
+  executeCommand(cmdGetDiskPartion)
+    .then(parseValidateOutput)
+    .catch(function (err) {
+      console.error(err)
+      D.failure(D.errorType.GENERIC_ERROR)
+    })
+}
+
+function parseValidateOutput (output) {
+  if (output && typeof output === 'string' && output.trim() !== '') {
```

Comment:
does that mean that as long as the device has any data for the command (even command not found) it will be identified as capable of collecting the data/